ui/job-view/redux/stores/pushes.js

```diff
@@ -0,0 +1,385 @@
+    max(Object.values(jobMap).map(job => new Date(`${job.last_modified}Z`))) ||
+    new Date();
+
+  latest.setSeconds(latest.getSeconds() - 3);
+  return latest;
+};
+
+/**
+ * Loops through the map of unclassified failures and checks if it is
+ * within the enabled tiers and if the job should be shown. This essentially
+ * gives us the difference in unclassified failures and, of those jobs, the
+ * ones that have been filtered out
+ */
+const doRecalculateUnclassifiedCounts = jobMap => {
+  const filterModel = new FilterModel();
+  const tiers = filterModel.urlParams.tier;
+  let allUnclassifiedFailureCount = 0;
+  let filteredUnclassifiedFailureCount = 0;
+
+  Object.values(jobMap).forEach(job => {
+    if (isUnclassifiedFailure(job)) {
+      if (tiers.includes(String(job.tier))) {
+        // if (filterModel.showJob(job)) {
+        if (filterModel.showJob(job)) {
+          filteredUnclassifiedFailureCount++;
+        }
+        allUnclassifiedFailureCount++;
+      }
+    }
+  });
+  return {
+    allUnclassifiedFailureCount,
+    filteredUnclassifiedFailureCount,
+  };
+};
+
+const addPushes = (data, pushList, jobMap, setFromchange) => {
+  if (data.results.length > 0) {
+    const pushIds = pushList.map(push => push.id);
+    const newPushList = [
+      ...pushList,
+      ...data.results.filter(push => !pushIds.includes(push.id)),
+    ];
+
+    newPushList.sort((a, b) => b.push_timestamp - a.push_timestamp);
+    const oldestPushTimestamp =
+      newPushList[newPushList.length - 1].push_timestamp;
+
+    const newStuff = {
+      pushList: newPushList,
+      oldestPushTimestamp,
+      ...doRecalculateUnclassifiedCounts(jobMap),
+      ...getRevisionTips(newPushList),
+    };
+
+    // since we fetched more pushes, we need to persist the push state in the URL.
+    const updatedLastRevision = newPushList[newPushList.length - 1].revision;
+
+    if (setFromchange && getUrlParam('fromchange') !== updatedLastRevision) {
+      const params = getAllUrlParams();
+      params.set('fromchange', updatedLastRevision);
+      replaceLocation(params);
+      // We are silently updating the url params, but we still want to
+      // update the ActiveFilters bar to this new change.
+      window.dispatchEvent(new CustomEvent(thEvents.filtersUpdated));
+    }
+
+    return newStuff;
+  }
+  return {};
+};
+
+const fetchNewJobs = () => {
+  return (dispatch, getState) => {
+    const {
+      pushes: { pushList, jobMap },
+    } = getState();
+
+    if (!pushList.length) {
+      // If we have no pushes, then no need to get jobs.
+      return;
+    }
+
+    const pushIds = pushList.map(push => push.id);
+    const lastModified = getLastModifiedJobTime(jobMap);
+
+    JobModel.getList(
+      {
+        push_id__in: pushIds.join(','),
+        last_modified__gt: lastModified.toISOString().replace('Z', ''),
+      },
+      { fetch_all: true },
+    ).then(data => {
+      const failureStatus = false;
+      if (!failureStatus) {
```

Comment:
Maybe I'm missing something but how would `failureStatus` ever be true? :) Would be be too much extra code to change the `JobModel` to use the `fetchData` wrapper in this pr? BTW, there's a `processResponse` helper in the `helpers/http.js` file for use with the getData wrapper - cuts down on some boilerplate.

Reply:
LOL...  holy cow.  :)  Yeah, my bad.  I meant to come back to this and somehow totally missed it.  Thank you for catching this.  It had seemed like maybe going too far to change this to use ``getData`` in this PR.  But perhaps not.  I'll go ahead and make that change.